Sources/Datadog/RUM/_RUMInternalProxy.swift

```diff
@@ -16,14 +16,31 @@ import Foundation
 /// are not considered part of the public interface of the Datadog SDK.
 public class _RUMInternalProxy {
     weak var subscriber: RUMCommandSubscriber?
+    private let dateProvider: DateProvider
 
-    init(subscriber: RUMCommandSubscriber) {
+    init(subscriber: RUMCommandSubscriber, dateProvider: DateProvider) {
         self.subscriber = subscriber
+        self.dateProvider = dateProvider
     }
 
-    public func addLongTask(at: Date, duration: TimeInterval, attributes: [AttributeKey: AttributeValue] = [:]) {
-        let longTaskCommand = RUMAddLongTaskCommand(time: at, attributes: attributes, duration: duration)
+    public func addLongTask(duration: TimeInterval, attributes: [AttributeKey: AttributeValue] = [:]) {
+        let longTaskCommand = RUMAddLongTaskCommand(time: dateProvider.now, attributes: attributes, duration: duration)
```

Comment:
Won't this (lack of `Date`) cause a skew in LTs reporting due to JS <> Swift dispatch time? I wonder on the motivation for dropping `Date` parameter 🤔💭.

Reply:
If we relied on the date from the RN SDK we would not be able to have the time sync with Kronos

Reply:
I know it's outdated diff, but just to give more context on NTP correction: 
* the correction is not applied automatically for `dateProvider.now`;
* instead, each product shifts dates individually when creating events;
* in case of LT and RUM, it happens in `RUMViewScope`, [here](https://github.com/DataDog/dd-sdk-ios/blob/b1838f7371a9c6ea2a073c8941e61c7f082c8fbe/Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift#L573).

By design, RUM ensures correct dates for all events originated in `*Command` signals. Because we used `RUMAddLongTaskCommand` here, that means the date would be corrected properly 👍.